<?xml version="1.0" ?>
<launch>
  <!-- Launch file for DS controller -->
  <!-- <arg name="ROBOT_NAME"                         default="icubSim" /> -->
  <arg name="TOPIC_OBJECT_POSE"           default="/simo_track/object_pose"/>
  <arg name="TOPIC_LEFT_ROBOT_BASE_POSE"  default="/simo_track/robot_left/pose"/>
  <arg name="TOPIC_LEFT_ROBOT_EE_POSE"    default="/simo_track/robot_left/ee_pose"/>
  <arg name="TOPIC_LEFT_ROBOT_COMMANDS"   default="/iiwa1/CustomControllers/command"/>

  <arg name="TOPIC_RIGHT_ROBOT_BASE_POSE" default="/simo_track/robot_right/pose"/>
  <arg name="TOPIC_RIGHT_ROBOT_EE_POSE"   default="/simo_track/robot_right/ee_pose"/>
  <arg name="TOPIC_RIGHT_ROBOT_COMMANDS"  default="/iiwa_blue/CustomControllers/command"/>

  <!-- args for iiwa_gazebo -->
  <arg name="robot_name_1"    default="iiwa1"/>
  <arg name="robot_name_2"    default="iiwa_blue"/>
  <arg name="iiwa_model_1"    default="7"/>
  <arg name="iiwa_model_2"    default="14"/>
  <arg name="force_sensor_1"  default="True"/>
  <arg name="grabber_1"       default="True"/>
  <arg name="force_sensor_2"  default="True"/>
  <arg name="grabber_2"       default="True"/>
  <arg name="controller"      default="CustomControllers"/>
  <arg name="position_1"      default="-y -1.5"/>
  <arg name="position_2"      default="-y 0.5 -x 0.00"/>

  <include file="$(find iiwa_gazebo)/launch/iiwa_double_gazebo.launch">
    <arg name="robot_name_1"    value="$(arg robot_name_1)" />
    <arg name="robot_name_2"    value="$(arg robot_name_2)"/>
    <arg name="iiwa_model_1"    value="$(arg iiwa_model_1)"/>
    <arg name="iiwa_model_2"    value="$(arg iiwa_model_2)"/>
    <arg name="force_sensor_1"  value="$(arg force_sensor_1)"/>
    <arg name="grabber_1"       value="$(arg grabber_1)"/>
    <arg name="force_sensor_2"  value="$(arg force_sensor_2)"/>
    <arg name="grabber_2"       value="$(arg grabber_2)"/>
    <arg name="controller"      value="$(arg controller)"/>
    <arg name="position_1"      value="$(arg position_1)"/>
    <arg name="position_2"      value="$(arg position_2)"/>
  </include>

  <!-- <include file="$(find sim_models_poses)/launch/sim_models_poses.launch">
  </include> -->

  <!-- <include file="$(find optitrack_subscriber)/launch/hardwares.launch">
      <arg name="robot_name_1" value="$(arg robot_name_1)"/>
      <arg name="robot_name_2" value="$(arg robot_name_2)"/>
  </include> -->
  
  <!-- <include file="$(find dual_arm_control)/launch/dual_arm_control.launch">
    <arg name="topic_pose_object"             value="$(arg TOPIC_OBJECT_POSE)"             />
    <arg name="topic_pose_robot_base_left"    value="$(arg TOPIC_LEFT_ROBOT_BASE_POSE)"    />
    <arg name="topic_pose_robot_ee_left"      value="$(arg TOPIC_LEFT_ROBOT_EE_POSE)"      />
    <arg name="topic_ee_commands_left"        value="$(arg TOPIC_LEFT_ROBOT_COMMANDS)"     />
    <arg name="topic_pose_robot_base_right"   value="$(arg TOPIC_RIGHT_ROBOT_BASE_POSE)"   />
    <arg name="topic_pose_robot_ee_right"     value="$(arg TOPIC_RIGHT_ROBOT_EE_POSE)"     />
    <arg name="topic_ee_commands_right"       value="$(arg TOPIC_RIGHT_ROBOT_COMMANDS)"    /> 
  </include> -->


</launch>